dbms/src/Storages/KVStore/FFI/ProxyFFI.cpp

```diff
@@ -1059,9 +1059,10 @@ void ReportThreadAllocateBatch(
 {
     try
     {
-        UNUSED(server);
         UNUSED(tid);
-        KVStore::reportThreadAllocBatch(buffToStrView(name), data);
+        if (!server || !server->tmt || !server->tmt->getKVStore())
+            return;
+        server->tmt->getKVStore()->reportThreadAllocBatch(buffToStrView(name), data);
```

Comment:
I think it is not necessary to keep another pointer in `KVStore`. Can we totally remove the `KVStore::joint_memory_allocation_map` and use `server->tmt->getContext()-> getJointThreadInfoJeallocMap()->...`?

Reply:
Two usage:
1. KVStore could be a separate module, just like DeltaMerge or PageStorage. I think it is better that all of them have its own `joint_memory_allocation_map`. When assembled as a whole, those `joint_memory_allocation_map` could be a pointer to the `joint_memory_allocation_map` in `Context`. This brings more benefits than making it easier to test each components, because if we want to do better later, we can create different instance of some enhanced version of `joint_memory_allocation_map`, which can trace memory allocations respectively for all components.
2. Furthermore, the pointer stored in KVStore is a "lifetime" declaration. Though `KVStore` is indirectly hold by `Context`, it only hold as a shared_ptr, not a unique_ptr that implies wholly ownership. Because `joint_memory_allocation_map` servers also Proxy, we must make sure the structure overlives Proxy, thus overliving KVStore. It is the simplest way.